tests/api/v2/handlers/test_abilities_api.py

```diff
@@ -0,0 +1,142 @@
+        assert resp.status == HTTPStatus.OK
+        ability = (await BaseService.get_service('data_svc').locate('abilities', {'ability_id': '123'}))[0]
+        assert ability.name == payload['name']
+        assert ability.tactic == payload['tactic']
+        assert ability.ability_id == test_ability.ability_id
+        assert ability.description == test_ability.description
+        assert ability.technique_id == test_ability.technique_id
+        assert ability.technique_name == test_ability.technique_name
+
+    async def test_unauthorized_update_ability(self, api_v2_client, test_ability):
+        payload = dict(name='an updated test ability', tactic='defense-evasion')
+        resp = await api_v2_client.patch('/api/v2/abilities/123', json=payload)
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_update_nonexistent_ability(self, api_v2_client, api_cookies):
+        payload = dict(name='an updated test ability', tactic='defense-evasion')
+        resp = await api_v2_client.patch('/api/v2/abilities/999', cookies=api_cookies, json=payload)
+        assert resp.status == HTTPStatus.NOT_FOUND
+
+    async def test_replace_ability(self, api_v2_client, api_cookies, test_ability):
+        test_executor_linux = Executor(name='sh', platform='linux', command='whoami')
+        payload = dict(name='replaced test ability', tactic='collection', technique_name='discovery', technique_id='2',
+                       executors=[ExecutorSchema().dump(test_executor_linux)])
+        resp = await api_v2_client.put('/api/v2/abilities/123', cookies=api_cookies, json=payload)
+        assert resp.status == HTTPStatus.OK
+        ability = await resp.json()
+        assert ability['name'] == payload['name']
+        assert ability['technique_name'] == payload['technique_name']
+        assert ability['technique_id'] == payload['technique_id']
+        assert ability['tactic'] == payload['tactic']
+        assert ability['ability_id'] == test_ability.ability_id
+
+    async def test_unauthorized_replace_ability(self, api_v2_client, test_ability):
+        test_executor_linux = Executor(name='sh', platform='linux', command='whoami')
+        payload = dict(name='replaced test ability', tactic='collection', technique_name='discovery', technique_id='2',
+                       executors=[ExecutorSchema().dump(test_executor_linux)])
+        resp = await api_v2_client.put('/api/v2/abilities/123', json=payload)
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_replace_nonexistent_ability(self, api_v2_client, api_cookies):
+        test_executor_linux = Executor(name='sh', platform='linux', command='whoami')
+        payload = dict(name='replaced test ability', tactic='collection', technique_name='discovery', technique_id='2',
+                       executors=[ExecutorSchema().dump(test_executor_linux)])
+        resp = await api_v2_client.put('/api/v2/abilities/123', cookies=api_cookies, json=payload)
```

Comment:
It looks like you're replacing an existing ability instead of a nonexisting one

Reply:
For this one I didn't include the test_ability fixture so the ability doesn't exist, but I think that's a bit confusing. I'll change the ID so it's clear that it is creating a new ability